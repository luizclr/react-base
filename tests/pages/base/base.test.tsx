import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";

import Base from "~/pages/base/base";
import initialStyleGuide from "~/style-guide";

describe("<Base />", () => {
  it("should be able to mount <Base /> component", () => {
    // given
    const { asFragment } = render(
      <BrowserRouter>
        <ThemeProvider theme={initialStyleGuide}>
          <Base />
        </ThemeProvider>
      </BrowserRouter>
    );

    // when
    const sut = asFragment();

    // then
    expect(sut).toMatchSnapshot();
  });
});
