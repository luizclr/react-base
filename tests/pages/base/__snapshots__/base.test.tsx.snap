// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should be able to mount <Base /> component 1`] = `
<DocumentFragment>
  .c1 {
  background-color: #1c1c1c;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 300px;
  border-right: 1px solid #383838;
  color: #e8e8e8;
  padding: 16px 0;
}

.c3 {
  padding: 0 16px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: inherit;
}

.c5 {
  list-style-type: none;
  margin-top: 32px;
}

.c6 {
  -webkit-transition: none;
  transition: none;
  font-weight: 500;
  border-bottom: 1px solid #383838;
  padding: 0;
}

.c6:hover {
  cursor: pointer;
  background-color: #383838;
}

.c6:last-child {
  border: none;
}

.c8 {
  -webkit-transition: none;
  transition: none;
  font-weight: 500;
  border-bottom: 1px solid #383838;
  padding: 16px;
}

.c8:hover {
  cursor: pointer;
  background-color: #383838;
}

.c8:last-child {
  border: none;
}

.c7 {
  color: #e8e8e8;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
}

.c9 {
  text-align: center;
  font-size: 14pt;
  margin-top: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
}

.c10 {
  display: block;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      data-testid="sidebar"
    >
      <h1
        class="c3"
      >
        React Base
      </h1>
      <div
        class="c4"
      >
        <ul
          class="c5"
        >
          <li
            class="c6"
          >
            <a
              class="c7"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="c6"
          >
            <a
              class="c7"
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="c8"
          >
            Logout
          </li>
        </ul>
        <div
          class="c1"
        >
          <ul
            class="c5"
          >
            <li
              class="c8"
            >
              Change theme
            </li>
          </ul>
          <p
            class="c9"
          >
            v0.1.0
          </p>
        </div>
      </div>
    </div>
    <div
      class="c10"
    />
  </div>
</DocumentFragment>
`;
